checkBalanced.js

```diff
@@ -0,0 +1,96 @@
+class TreeNode {
+    constructor(data) {
+        this.data = data
+        this.left = null
+        this.right = null
+    }
+}
+
+class Tree {
+    constructor() {
+        this.head = null
+    }
+
+    add(data) {
+        const newNode = new TreeNode(data)
+        if (!this.head) {
+            this.head = newNode
+            return
+        }
+        this.addNode(this.head, newNode)
+    }
+
+    addNode(node, newNode) {
+        if (newNode.data < node.data) {
+            if (!node.left) {
+                node.left = newNode
+            } else {
+                this.addNode(node.left, newNode)
+            }
+        } else {
+            if (!node.right) {
+                node.right = newNode
+            } else {
+                this.addNode(node.right, newNode)
+            }
+        }
+    }
+
+    print() {
+        if (!this.head) {
+            console.log("Empty Tree")
+            return
+        }
+        const queue = [this.head]
+        while (queue.length) {
+            const node = queue.shift()
+            process.stdout.write(node.data + " ")
```

Comment:
Is there a reason you're not using `console.log` here? It's a nice use of a stream, but this code wouldn't run on a browser or code sandbox.

Reply:
You should also use string interpolation instead of string addition in JS.
```suggestion
            console.log(`${node.data} `)
```